libsolidity/ast/Types.cpp

```diff
@@ -2354,14 +2355,26 @@ unsigned FunctionType::sizeOnStack() const
 	}
 
 	unsigned size = 0;
-	if (kind == Kind::External || kind == Kind::CallCode || kind == Kind::DelegateCall)
+
+	switch(kind)
+	{
+	case Kind::External:
+	case Kind::CallCode:
+	case Kind::DelegateCall:
 		size = 2;
-	else if (kind == Kind::BareCall || kind == Kind::BareCallCode || kind == Kind::BareDelegateCall)
-		size = 1;
-	else if (kind == Kind::Internal)
-		size = 1;
-	else if (kind == Kind::ArrayPush || kind == Kind::ByteArrayPush)
+		break;
+	case Kind::BareCall:
+	case Kind::BareCallCode:
+	case Kind::BareDelegateCall:
+	case Kind::Internal:
+	case Kind::ArrayPush:
+	case Kind::ByteArrayPush:
 		size = 1;
+		break;
+	default:
```

Comment:
It doesn't do that before the change. Also if the default is removed the compiler will complain that not all cases are handled.